src/tablecloth/time/api/converters.clj

```diff
@@ -20,6 +22,11 @@
 (defn year->local-date [^Year year]
   (-> year (.atMonthDay (java.time.MonthDay/parse "--01-01"))))
 
+(defn year-quarter->local-date [year-quarter format-pattern]
+  (-> year-quarter
+      (YearQuarter/parse (DateTimeFormatter/ofPattern format-pattern))
+      .atEndOfQuarter))
```

Comment:
I think that our convention should be instead to use the start of the quarter. Could be convinced otherwise, but that make more intuitive sense to me. If I think of the date representation of a quarter and a month, I wouldn't think of the end of the period... 

Or do you have a principled reason for choosing the end?

Reply:
I wanted to support both begin and end, but did not see the begin implementation. This is another reason why we should move to the 2nd option when doable - and we can remove dependencies on the threeten conventions.

I am thinking of refactoring so it fits with the approach you quoted above, but really treat it as a hack to move forward

Reply:
> I wanted to support both begin and end, but did not see the begin implementation. 

For these, see [here](https://github.com/scicloj/tablecloth.time/pull/41/files#diff-43bd2071876749f3f611695f45ed98fe9fa27f05f61bed4afbae429e87312e2dR5-R6) and [here](https://github.com/scicloj/tablecloth.time/pull/41/files#diff-142e91d65911a72c8ec9c8bf618cc6331c2bc02748888db6a8af739419eebd36R8-R9)

> I am thinking of refactoring so it fits with the approach you quoted above, but really treat it as a hack to move forward

To which approach are you referring here?

Reply:
Assuming you are referring to the atDay method as that puzzled me and was not sure what that means as the period spans 3 months. I was looking for the equivalent of .atEndOfQuarter for the beginning. Obviously we can shift the previous quarter value by 1 day, but did not have appetite to do that. So, I remained at end of period as a easy convention.

Refactoring from convertors so these "time objects that are higher than date" data types are available in utils. In particular, move the bi-directional conversion methods for year-quarter (both string->local-date, and local-date->string)

Reply:
> Assuming you are referring to the atDay method as that puzzled me and was not sure what that means as the period spans 3 months...

`(.atDay some-year-quarter 1)` just gives you the first day of the each quarter. E.g.
```clj
(-> "2021-Q1"
    org.threeten.extra.YearQuarter/parse
    (.atDay 1))
;; => #time/date "2021-01-01"

(-> "2021-Q2"
    org.threeten.extra.YearQuarter/parse
    (.atDay 1))
;; => #time/date "2021-04-01"

;; etc...
```